README.md

```diff
@@ -27,3 +27,15 @@ about the current recursive call.
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. You do not need to prove that the invariant is correct. Add
 your answer to this markdown file.
+
+## Answer
+
+The array contains the fibonacci values from (0, n) at the beginning of every iteration.
```

Comment:
But aren't you computing the nth Fibonacci number in that iteration? So your invariant is that the result must be available before it's computed?